### DISCUSSION TITLE
Only 1000 items can be fetched using fetch

### DISCUSSION THREAD

**Discussion by ghost** [deta/cloud-docs#224] [Help] [answered]
Can base register items after 1000?
If you fetch with an empty argument and refer to items, only 1000 items will be recognized

```py
from deta import Deta

deta = Deta("")
bots = deta.Base("bots")

print(len(bots.fetch().items))
```

>>> 1000

**Comment by jnsougata** [comment, marked as answer]
`fetch` has optional parameter `last` which is a string (last key evaluated in the response)
check `last` parameter in hare https://docs.deta.sh/docs/base/sdk/#fetch
that will help you to fetch all the items from deta base in case you have more than 1000 items. keep fetching data from from base with the `last` key from the response until the last key in the response is `None`

**Reply by ghost** [reply]
Thank you!
I succeeded in getting more than 1000.
